### PR TITLE
Array attributes first and last

### PR DESCRIPTION
**Pull request by xbauch**
As far as I could find there are only 4 array attributes and they all work for static arrays (at least to the extend of the added test). For dynamic arrays, this PR adds support for `first` and `last` (length worked before). `Range` for dynamic is still unsupported.

 I also include test for `in_expression` that uses range-attribute but fails since we do not support `n_in`.

### CODE REVIEW

File: testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+procedure Arrays_Attributes is
```
Comment (sonodtt) [inline testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb]: would be helpful to list the attributes at the top in a comment so people know at a glance what to look for:
e.g 
--  array attributes are: first, last, range, length

**Comment by hannes-steffenhagen-diffblue** [comment]
Can you also add a test for dynamic attributes, i.e.

```
procedure Test is 
  type Indet_Array is array(Integer range <>) of Integer;

  procedure Array_Consumer(Arr : Indet_Array) is 
  begin
    pragma Assert(Arr'First = 1);
  end Array_Consumer;

  My_Arr : Indet_Array(1..2) := (others => 0);
begin
  Array_Consumer(My_Arr);
end Test;
```

?

File: testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+   subtype Range1 is Integer range 10 .. 20;
+   type Arr3 is array (Range1) of Integer;
+   Actual3 : Arr3 := (7, 8, 9, others => 10);
+   subtype My_Index is Integer range Actual3'Range;
+   Array_Attribs : Integer;
+   Val : Integer := 10;
+   An_Index : My_Index := 12;
+begin
+   Actual3 (10 .. 11) := Actual3 (17 .. 18);
+   Actual3 (11) := 100;
```
Comment (sonodtt) [inline testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb]: After 'Actual3 (10 .. 11) := Actual3 (17 .. 18);'
why not a good old pragma 
Actual3(10)=10;

File: testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+   Actual3 (13 .. 16) := Actual3 (11 .. 12) & Actual3 (15 .. 16);
+   Array_Attribs := Actual3'First + Actual3'Last + Actual3'Length;
+
+   pragma Assert (Array_Attribs = 41);
+   pragma Assert (An_Index > 7);
```
Comment (sonodtt) [inline testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb]: An_index is not reassigned to so seeing this check not as =12 made me check twice if it was assigned to somewhere I'd missed.

File: testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb (hunk 1 above)
Comment (xbauch) [reply testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb]: Done.

File: testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb (hunk 3 above)
Comment (xbauch) [reply testsuite/gnat2goto/tests/arrays_attributes/arrays_attributes.adb]: Done.

**Pull request merged**